blockchain/utxocache.go

```diff
@@ -58,6 +58,59 @@ const (
 	periodicFlushInterval = time.Minute * 2
 )
 
+// UtxoCacher represents a utxo cache that sits on top of the utxo set database.
+// All utxo reads and writes MUST go through the cache, and never read or write
+// to the database directly.
+//
+// The interface contract requires that all of these methods are safe for
+// concurrent access.
+type UtxoCacher interface {
```

Comment:
Good call.  I removed the parts that are implementation dependent.  I was also considering removing the details around the underlying database, as it could potentially be some other persistent data store, but since currently one of the functions takes a database transaction as a parameter, I left that as-is for now.

Reply:
Looks good.  Thanks for the updates.